pkg/controller/certificates/util_test.go

```diff
@@ -282,67 +281,44 @@ func selfSignCertificate(t *testing.T, spec cmapi.CertificateSpec) []byte {
 	return pemData
 }
 
-func TestRenewBeforeExpiryDuration(t *testing.T) {
+func TestCalculateRenewalTime(t *testing.T) {
 	type testCase struct {
-		notBefore                        time.Time
-		notAfter                         time.Time
-		specRenewBefore                  *metav1.Duration
-		defaultRenewBeforeExpiryDuration time.Duration
-		expected                         time.Duration
+		notBefore       time.Time
+		notAfter        time.Time
+		specRenewBefore time.Duration
+		expected        time.Time
 	}
-	now := time.Now()
+
+	now := time.Date(2000, 6, 1, 0, 0, 0, 0, time.UTC)
+
 	tests := map[string]testCase{
-		"use default": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  nil,
-			defaultRenewBeforeExpiryDuration: time.Hour,
-			expected:                         time.Hour,
-		},
-		"spec overrides default": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  &metav1.Duration{Duration: time.Hour * 2},
-			defaultRenewBeforeExpiryDuration: time.Hour,
-			expected:                         time.Hour * 2,
+		"renewBefore larger than actual duration": {
+			notBefore:       now,
+			notAfter:        now.Add(time.Hour * 24),
+			specRenewBefore: time.Hour * 24 * 7,
+			// Fall back to 2/3 of actual duration
+			expected: now.Add(time.Hour * 24 * 2 / 3),
 		},
-		"default larger than actual duration": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  nil,
-			defaultRenewBeforeExpiryDuration: time.Hour * 24 * 7,
-			expected:                         time.Hour * 8,
+		"renewBefore equal to actual duration": {
+			notBefore:       now,
+			notAfter:        now.Add(time.Hour * 24),
+			specRenewBefore: time.Hour * 24,
+			// Fall back to 2/3 of actual duration
+			expected: now.Add(time.Hour * 24 * 2 / 3),
 		},
-		"spec larger than actual duration": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  &metav1.Duration{Duration: time.Hour * 24 * 7},
-			defaultRenewBeforeExpiryDuration: time.Hour,
-			expected:                         time.Hour * 8,
-		},
-		"default equal to actual duration": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  nil,
-			defaultRenewBeforeExpiryDuration: time.Hour * 24,
-			expected:                         time.Hour * 8,
-		},
-		"spec equal to actual duration": {
-			notBefore:                        now,
-			notAfter:                         now.Add(time.Hour * 24),
-			specRenewBefore:                  &metav1.Duration{Duration: time.Hour * 24},
-			defaultRenewBeforeExpiryDuration: time.Hour,
-			expected:                         time.Hour * 8,
```

Comment:
If we keep the method signature the same (returning time.Duration), it'd be great to see these unit tests left in place so we can be sure that the behaviour has not changed in an unexpected way (presumably this should only be net _new_ behaviour to fix bugs that happen in edge cases that we haven't covered with unit tests)